lib/api.js

```diff
@@ -239,6 +239,7 @@ function transformOptions(options, encoding) {
       runScripts: undefined,
       encoding,
       pretendToBeVisual: false,
+      storageToDisk: false,
```

Comment:
I'd avoid tying us down to files as a storage provider. Maybe it'd make more sense to allow a way to listen for storage update events and just provide those with domain + js object.

Reply:
Hmm, this was my interpretation of Domenic's comment in the issue - "For now we can keep everything in-memory, although eventually we should provide a way of storing to disk if people want that".

If we want something more complex I think I'll focus on just the internal implementation for this pull request, and let us decide on this later.

Reply:
Right, I'd say for the initial implementation, both local and session storage are in-memory, and in the future, we could allow providing some kind of storage provider interface (which could also be used for HTTP caches, maybe!).